### PR TITLE
Added frontend and middleware manifest files from Online Boutique v0.3.5

### PR DESCRIPTION
**Pull request by alizaidis**
Added frontend and middleware manifest files from Online Boutique v0.3.5 to be consumed in tutorial showing a migration path from a single cluster deployed version to multiple clusters using multi-cluster Services and Gateways.

### CODE REVIEW

File: gateway/docs/onlineboutique-frontend-manifests.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  ports:
+  - name: http
+    port: 80
+    targetPort: 8080
+---
+kind: ServiceExport
+apiVersion: net.gke.io/v1
+metadata:
+ namespace: onlineboutique
+ name: frontend
```
Comment (googledrew) [inline gateway/docs/onlineboutique-frontend-manifests.yaml:94]: not a huge deal, but... no newline.

File: gateway/docs/onlineboutique-middleware-manifests.yaml [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+metadata:
+  name: adservice
+spec:
+  type: ClusterIP
+  selector:
+    app: adservice
+  ports:
+  - name: grpc
+    port: 9555
+    targetPort: 9555
```
Comment (googledrew) [inline gateway/docs/onlineboutique-middleware-manifests.yaml:477]: not a huge deal, but... no newline.

File: gateway/docs/onlineboutique-frontend-manifests.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
+##################################################
+# Referenced by [Link TBD]
+##################################################
```
Comment (googledrew) [inline gateway/docs/onlineboutique-frontend-manifests.yaml:3]: Is the license header not required?

File: gateway/docs/onlineboutique-middleware-manifests.yaml [hunk 4]
```diff
@@ -0,0 +1,477 @@
+##################################################
+# Referenced by [Link TBD]
+##################################################
```
Comment (googledrew) [inline gateway/docs/onlineboutique-middleware-manifests.yaml:3]: license header?

**Review by googledrew — APPROVED** [review]

**Review by googledrew — APPROVED** [review]

**Review by googledrew — APPROVED** [review]

File: gateway/docs/cluster-migration/onlineboutique-frontend-manifests.yaml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+          - name: CART_SERVICE_ADDR
+            value: "cartservice.onlineboutique.svc.clusterset.local:7070"
+          - name: RECOMMENDATION_SERVICE_ADDR
+            value: "recommendationservice:8080"
+          - name: SHIPPING_SERVICE_ADDR
+            value: "shippingservice:50051"
+          - name: CHECKOUT_SERVICE_ADDR
+            value: "checkoutservice:5050"
+          - name: AD_SERVICE_ADDR
+            value: "adservice:9555"
```
Comment (boredabdel) [inline gateway/docs/cluster-migration/onlineboutique-frontend-manifests.yaml:70]: can you please ensure internal service are consistently referenced here. Either with the full FQDN or without. Just make it's consistent across all the env variables.

File: gateway/docs/cluster-migration/onlineboutique-frontend-manifests.yaml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      serviceAccountName: default
+      containers:
+        - name: server
+          image: gcr.io/google-samples/microservices-demo/frontend:v0.3.5
+          ports:
+          - containerPort: 8080
+          readinessProbe:
+            initialDelaySeconds: 10
+            httpGet:
+              path: "/_healthz"
```
Comment (boredabdel) [inline gateway/docs/cluster-migration/onlineboutique-frontend-manifests.yaml:41]: If i understand it correctly from the all the microservices of the online boutique this is the only one which is exposed to the outside ? If that's the case , don't you need a BackendConfig object to customize the HealthCheck Path?

File: gateway/docs/cluster-migration/onlineboutique-middleware-manifests.yaml [hunk 7]
```diff
@@ -0,0 +1,491 @@
...
+          - name: SHIPPING_SERVICE_ADDR
+            value: "shippingservice:50051"
+          - name: PAYMENT_SERVICE_ADDR
+            value: "paymentservice:50051"
+          - name: EMAIL_SERVICE_ADDR
+            value: "emailservice:5000"
+          - name: CURRENCY_SERVICE_ADDR
+            value: "currencyservice:7000"
+          - name: CART_SERVICE_ADDR
+            value: "cartservice.onlineboutique.svc.clusterset.local:7070"
```
Comment (boredabdel) [inline gateway/docs/cluster-migration/onlineboutique-middleware-manifests.yaml:114]: Same comment, just make sure internal services are referenced in a consistent way please.

**Review by boredabdel — CHANGES_REQUESTED** [review]
Few comments.

**Review by mark-church — APPROVED** [review]

File: gateway/docs/cluster-migration/onlineboutique-frontend-manifests.yaml (hunk 5 above)
Comment (mark-church) [reply gateway/docs/cluster-migration/onlineboutique-frontend-manifests.yaml:70]: agreed - makes sense to use cluster.local for the rest to differentiate

Comment (alizaidis) [reply gateway/docs/cluster-migration/onlineboutique-frontend-manifests.yaml]: Updated service hostnames to be consistent.

File: gateway/docs/cluster-migration/onlineboutique-middleware-manifests.yaml (hunk 7 above)
Comment (alizaidis) [reply gateway/docs/cluster-migration/onlineboutique-middleware-manifests.yaml]: Updated service hostnames to be consistent.

File: gateway/docs/cluster-migration/onlineboutique-frontend-manifests.yaml (hunk 6 above)
Comment (alizaidis) [reply gateway/docs/cluster-migration/onlineboutique-frontend-manifests.yaml:41]: Currently the BackendConfig resource is not supported for multi-cluster GatewayClasses (https://cloud.google.com/kubernetes-engine/docs/how-to/deploying-gateways#limitations) which we are using in the reference guide consuming these manifests. The app responds on port 80 for health as well, so this implementation does work in its current form. Happy to update with BackendConfig when it is supported for multi-cluster GatewayClasses if that works.

**Review by boredabdel — APPROVED** [review]
LGTM

**Pull request merged**